engine/runtime/src/main/scala/org/enso/interpreter/runtime/util/DiagnosticFormatter.scala

```diff
@@ -25,17 +26,46 @@ class DiagnosticFormatter(
   private val maxSourceLinesToPrint          = 3
   private val linePrefixSize                 = blankLinePrefix.length
   private val outSupportsAnsiColors: Boolean = outSupportsColors
-  private val (textAttrs: fansi.Attrs, subject: String) = diagnostic match {
-    case _: Error   => (fansi.Color.Red ++ fansi.Bold.On, "error: ")
-    case _: Warning => (fansi.Color.Yellow ++ fansi.Bold.On, "warning: ")
+
+  sealed protected trait DiagnosticKind
+
+  protected object DiagnosticKind {
```

Comment:
Wouldn't it be better to switch to Java and just do:
```java
enum DiagnosticKind {
  ERROR, WARNING;
}
```
?

Just a note of despaired reader of Scala code....

Reply:
Yeah, I think you are right...